cas-server-core/src/main/java/org/jasig/cas/services/JsonServiceRegistryConfigWatcher.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import static java.nio.file.StandardWatchEventKinds.*;
+
+/**
+ * This is {@link JsonServiceRegistryConfigWatcher} that watches the json config directory
+ * for changes and promptly attempts to reload the CAS service registry configuration.
+ *
+ * @author Misagh Moayyed
+ * @since 4.1.0
+ */
+class JsonServiceRegistryConfigWatcher implements Runnable {
```

Comment:
No that's different. What that does is, it periodically reloads the configuration from the source regardless. What I am proposing is, configuration should be reloaded as soon as a change occurs, so you wouldn't have to wait 5 mins (or whatever the timeout is). You make it a change and it's instant. 

Suppose app owner X comes to you and says "hey we need attribute X for our app", and your registry is set to reload once every 1 hr. You cant tell him to wait an hour until the change is reloaded. You'd rather make the change and have him test it immediately.